### PR TITLE
MueLu/Ifpack2/Tpetra: Adding Reitzinger / Hiptmair support / Fixing SpGEMM bug

### PR DESCRIPTION
**Pull request by csiefer2**
Issue: #9506

Stakeholder feedback: n/a (non-ASC)

TESTS:  Yup!  Added new tests to MueLu

### CODE REVIEW

**Comment by jhux2** [comment]
I recommend a big old can of squash.

**Comment by csiefer2** [comment]
Failing clang tests cannot be reproduced.  Ugh. @trilinos/framework

**Comment by csiefer2** [comment]
Slapping 'on retest.

**Comment by lucbv** [comment]
> I recommend a big old can of squash.

I recommend splitting this PR by package so each new PR can be reviewed appropriately and the number of issues/bugs in each PR is lower and easier to assess and fix!

**Comment by csiefer2** [comment]
> 
> 
> > I recommend a big old can of squash.
> 
> I recommend splitting this PR by package so each new PR can be reviewed appropriately and the number of issues/bugs in each PR is lower and easier to assess and fix!

No way.

File: packages/muelu/src/Smoothers/MueLu_Ifpack2Smoother_def.hpp [hunk 1]
```diff
@@ -196,6 +196,15 @@ namespace MueLu {
       // Aggregate smoothing needs aggregates
       this->Input(currentLevel,"Aggregates");
     }
+    else if (type_ == "HIPTMAIR") {
+
+      currentLevel.print(std::cout,Debug);//CMS
```
Comment (cgcgcg) [inline packages/muelu/src/Smoothers/MueLu_Ifpack2Smoother_def.hpp:201]: Remove?

File: cmake/std/PullRequestLinuxCuda10.1.105uvmOffTestingSettings.cmake [hunk 2]
```diff
@@ -247,5 +247,8 @@ set (SEACASIoss_exodus_fpp_serialize_DISABLE ON CACHE BOOL "Turn off tests for n
 # Stratimikos UVM = OFF tests
 set (Stratimikos_test_single_amesos2_tpetra_solver_driver_KLU2_MPI_1_DISABLE ON CACHE BOOL "Turn off tests for non-UVM build")
 
+# MueLu UVM = OFF tests
+set (MueLu_ReitzingerPFactory_MPI_4_DISABLE ON CACHE BOOL "Turn off tests for non-UVM build")
+set (MueLu_Maxwell3D-Tpetra_2_MPI_4_DISABLE ON CACHE BOOL "Turn off tests for non-UVM build")
```
Comment (cgcgcg) [inline cmake/std/PullRequestLinuxCuda10.1.105uvmOffTestingSettings.cmake:252]: Do you still need to turn off tests for the no-UVM build?

**Review by cgcgcg — DISMISSED** [review]

File: cmake/std/PullRequestLinuxCuda10.1.105uvmOffTestingSettings.cmake (hunk 2 above)
Comment (csiefer2) [reply cmake/std/PullRequestLinuxCuda10.1.105uvmOffTestingSettings.cmake:252]: Yes.  Until Jonathan's PR merges.

File: packages/muelu/src/Smoothers/MueLu_Ifpack2Smoother_def.hpp (hunk 1 above)
Comment (csiefer2) [reply packages/muelu/src/Smoothers/MueLu_Ifpack2Smoother_def.hpp]: Yup.  Fixed.

**Review by cgcgcg — APPROVED** [review]

**Pull request merged**